### DISCUSSION TITLE
insert with primary key value (database)

### DISCUSSION THREAD

**Discussion by devpro2020** [supabase/supabase#18637] [Questions]
i cannot insert any data in to my table after inserting data with primary key field. How can i fix this. I have a table lessons and id (primary, is identity) column. If i try insert with empty id column Error message is: duplicate key value violates unique constraint "lessons_pkey"

**Comment by GaryAustin1** [comment]
If you are using an identity column you pretty much never want to insert with the id in the data.  If you do and the sequence (which does not look at the data in the table) will eventually hit your id's you have put in there.   You need to set your sequence now higher than your highest used id for it not to conflict again.  `SELECT setval('tablename_id_seq', 21);  `  where tablename_id_seq where table name is the usual name for the sequence with an id column.

https://github.com/supabase/supabase/issues/1804
